tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+                "mksrf_fvocef     = "+self.input_path+"mksrf_vocef_0.5x0.5_simyr2000.c110531.nc" +"\n"
+                "mksrf_furbtopo   = "+self.input_path+"mksrf_topo.10min.c080912.nc"+"\n"
+                "mksrf_fgdp       = "+self.input_path+"mksrf_gdp_0.5x0.5_AVHRR_simyr2000.c130228.nc"+"\n"
+                "mksrf_fpeat      = "+self.input_path+"mksrf_peatf_0.5x0.5_AVHRR_simyr2000.c130228.nc"+"\n"
+                "mksrf_fsoildepth = "+self.input_path+"mksf_soilthk_5x5min_ORNL-Soil_simyr1900-2015_c170630.nc"+"\n"
+                "mksrf_fabm       = "+self.input_path+"mksrf_abm_0.5x0.5_AVHRR_simyr2000.c130201.nc"+"\n"
+                "outnc_double     = .true. \n"
+                "all_urban        = .false.\n"
+                "no_inlandwet     = .true. \n"
+                "mksrf_furban     = "+self.input_path+"mksrf_urban_0.05x0.05_simyr2000.c170724.nc"+"\n"
+                "gitdescribe      = "+label+"\n"
+                "mksrf_ftopostats = "+self.input_path+"mksrf_topostats_1km-merge-10min_HYDRO1K-merge-nomask_simyr2000.c130402.nc"+"\n"
+                "mksrf_fvegtyp    = "+self.input_path + "pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629/mksrf_landuse_histclm50_LUH2_1850.c170629.nc"+"\n"
+                "mksrf_fsoicol    = "+self.input_path + "pftcftlandusedynharv.0.25x0.25.MODIS.simyr1850-2015.c170412/mksrf_soilcolor_CMIP6_simyr2005.c170623.nc"+"\n"
+                "mksrf_flai       = "+self.input_path + "pftcftlandusedynharv.0.25x0.25.MODIS.simyr1850-2015.c170412/mksrf_lai_78pfts_simyr2005.c170413.nc" +"\n"
+                "fdyndat          = ''\n"
+                "numpft           = "+self.num_pft+"\n"
+                "dst_mesh_file    = "+self.input_path+dst_mesh+"\n"
+                "\n&transient\n"
+                "use_transient    = "+use_transient + "\n"
+                "start_year       = "+self.start_year.__str__() + "\n"
+                "end_year         = "+self.end_year.__str__() + "\n"
+                "mksrf_dyn_lu     = "+self.input_path+ "pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629" + "\n"
+                "mksrf_fdynuse    = "+self.lu_fname + "\n"
+                "\n&vic\n"
+                "use_vic          = "+self.vic_flag.__str__()+"\n"
+                "mksrf_fvic       = "+self.input_path+"mksrf_vic_0.9x1.25_GRDC_simyr2000.c130307.nc\n"
+                "outnc_vic        = \n"
+                "\n&glc\n"
+                "use_glc          = "+self.glc_flag.__str__()+"\n"
+                "outnc_3dglc      = \n"
+                "/\n"
+                )
+
+        print ("Successfully created namelist file : ", self.namelist_fname,".")
+        print ("--------------------------------------------------------")
+        namelist_file.write (nl_template)
+        namelist_file.close ()
+
+def which_mesh(res):
+   '''
+   Figure out the dst mesh file for each res
+   '''
+   switcher = { 
```

Comment:
Better to read the .xml file that contains the same info or is there a plan to remove this info from the .xml file?